ethcore/src/blockchain/blockchain.rs

```diff
@@ -571,8 +571,8 @@ impl BlockChain {
 
 		{
 			// Fetch best block details
-			let best_block_total_difficulty = bc.block_details(&best_block_hash).unwrap().total_difficulty;
-			let best_block_rlp = bc.block(&best_block_hash).unwrap();
+			let best_block_total_difficulty = bc.block_details(&best_block_hash).expect("Best block is from a known block hash; a known block hash always comes with a known block detail; qed").total_difficulty;
```

Comment:
This might benefit from breaking across lines, though not sure what our convention is here. There is some precedent of having `expect` on the following line in other parts of the codebase.